src/client.rs

```diff
@@ -0,0 +1,557 @@
+
+                let mut stream = match client.subscribe_execution_payloads_v2(req).await {
+                    Ok(stream) => stream.into_inner(),
+                    Err(e) => {
+                        tracing::error!(error = ?e, "Error in execution payload stream, retrying...");
+                        tokio::time::sleep(Duration::from_secs(2)).await;
+                        continue;
+                    }
+                };
+
+                tracing::info!("Execution payload stream established");
+
+                while let Some(item) = stream.next().await {
+                    match item {
+                        Ok(payload) => {
+                            let payload_deserialized = match payload.data_version {
+                                BELLATRIX_DATA_VERSION => {
+                                    ExecutionPayloadV1::from_ssz_bytes(&payload.ssz_payload)
+                                        .map(ExecutionPayload::V1)
+                                }
+                                CAPELLA_DATA_VERSION => {
+                                    ExecutionPayloadV2::from_ssz_bytes(&payload.ssz_payload)
+                                        .map(ExecutionPayload::V2)
+                                }
+                                DENEB_DATA_VERSION => {
+                                    ExecutionPayloadV3::from_ssz_bytes(&payload.ssz_payload)
+                                        .map(ExecutionPayload::V3)
+                                }
+                                _ => {
+                                    tracing::error!(
+                                        data_version = payload.data_version,
+                                        "Error deserializing execution payload: invalid data version"
+                                    );
+                                    continue;
+                                }
+                            };
+
+                            match payload_deserialized {
+                                Ok(payload) => tx.send(payload).ok(),
+                                Err(e) => {
+                                    tracing::error!(error = ?e, "Error deserializing execution payload");
+                                    continue;
+                                }
+                            };
+                        }
+                        Err(e) => {
+                            tracing::error!(error = ?e, "Error in execution payload stream, retrying...");
+                            // If we get an error, we set the inner stream to None and break out of the loop.
+                            // Next iteration will retry the stream.
+                            break;
+                        }
+                    }
+                }
+            }
+        });
+
+        UnboundedReceiverStream::new(rx)
+    }
+
+    pub async fn subscribe_new_raw_execution_payloads(&self) -> impl Stream<Item = Bytes> {
```

Comment:
I think we can leave this out. Most users won't decode SSZ execution payloads themselves, but they might do it for full beacon blocks (like below)

Reply:
@mempirate I agree, should I also removed it on fiber-go?

Reply:
Yeah let's do that